.github/workflows/main.yml

```diff
@@ -58,13 +58,23 @@ jobs:
         uses: guyarb/golang-test-annotations@v0.5.1
         with:
           test-results: test_results.json
-
+      - name: Run static check
+        # Only run if the test failed on target version to avoid duplicated annotations on GitHub.
+        if: ${{ always() && env.TARGET_GOLANG_VERSION == matrix.go }}
+        uses: reviewdog/action-staticcheck@v1
+        # with:
+        #   # TODO: pick filter https://github.com/reviewdog/reviewdog#filter-mode
+        #   filter_mode: nofilter
+      - name: Run golangci-lint
+        # Only run if the test failed on target version to avoid duplicated annotations on GitHub.
+        if: ${{ always() && env.TARGET_GOLANG_VERSION == matrix.go }}
+        uses: golangci/golangci-lint-action@v3
       # TODO(@okdas): move coverage to a separate job
       - name: create coverage report
-        if: ${{ env.TARGET_GOLANG_VERSION == matrix.go }}
+        if: ${{ always() && env.TARGET_GOLANG_VERSION == matrix.go }}
```

Comment:
Why did we add `always()` here?

Reply:
> Why did we add `always()` here?

`always()` forces step to run even if previous step(s) failed. 

https://docs.github.com/en/actions/learn-github-actions/expressions#always

Reply:
Cool. Ty for the link and explanation!